### ISSUE TITLE
[FEATURE REQ] Add Spring Cloud Azure starter when users choose MySQL/PostgreSQL and Azure on Spring Initializer

### ISSUE THREAD

**Issue by saragluna** [Azure/azure-sdk-for-java#30768]
**Is your feature request related to a problem? Please describe.**

We now have two Spring starters for supporting AAD authentication:

  - `spring-cloud-azure-starter-jdbc-mysql`
  - `spring-cloud-azure-starter-jdbc-postgresql`

We should add these two starters when users choose **MySQL/PostgreSQL Driver** + **Azure Support** on Spring Initializer:

 *Scene 1*:  choose **MySQL Driver** + **Azure Support**
 
1. **Status quo**: 

 Currently, when the user selects **MySQL Driver** and **Azure Support** in the Spring Initializer:

   1.1 The generated `pom.xml` will contain the following:

   ```XML
       <dependency>
         <groupId>com.azure.spring</groupId>
         <artifactId>spring-cloud-azure-starter</artifactId>
       </dependency>

       <dependency>
         <groupId>com.mysql</groupId>
         <artifactId>mysql-connector</artifactId>
         <scope>runtime</scope>
       </dependency>
   ```

   1.2 The contents of `HELP.md` are as follows:

   <img width="337" alt="image" src="https://user-images.githubusercontent.com/58474919/206621452-e6f764d1-c857-4768-b01f-6d268d110c52.png">

2. **Goal**

  When the user selects **MySQL Driver** and **Azure Support** in the Spring Initializer:

   2.1 The generated `pom.xml` will contain the following:

   ```XML
    <dependency>
        <groupId>com.azure.spring</groupId>
        <artifactId>spring-cloud-azure-starter-jdbc-mysql</artifactId>
    </dependency>
   ```

   2.2 Include a link to the [Azure MySQL reference documentation](https://learn.microsoft.com/azure/developer/java/spring-framework/spring-cloud-azure?tabs=maven#mysql-support) in the `Reference Documentation` section of `HELP.md`.

 *Scene 2*:  choose **PostgreSQL Driver** + **Azure Support**
 
1. **Status quo**: 

    Currently, when the user selects **PostgreSQL Driver** and **Azure Support** in the Spring Initializer:

   1.1 The generated `pom.xml` will contain the following:

   ```XML
       <dependency>
           <groupId>com.azure.spring</groupId>
           <artifactId>spring-cloud-azure-starter</artifactId>
       </dependency>

      <dependency>
          <groupId>org.postgresql</groupId>
          <artifactId>postgresql</artifactId>
          <scope>runtime</scope>
      </dependency>
   ```

   1.2 The contents of `HELP.md` are as follows:

   <img width="407" alt="image" src="https://user-images.githubusercontent.com/58474919/206627292-69717b7e-4d1d-4895-abea-9f798329359b.png">

2. **Goal**

     When the user selects **PostgreSQL Driver** and **Azure Support** in the Spring Initializer:

      2.1 The generated `pom.xml` will contain the following:

   ```XML
    <dependency>
        <groupId>com.azure.spring</groupId>
       <artifactId>spring-cloud-azure-starter-jdbc-postgresql</artifactId>
    </dependency>
   ```

   2.2 Add a link to the [Azure PostgreSQL reference documentation](https://learn.microsoft.com/azure/developer/java/spring-framework/spring-cloud-azure?tabs=maven#postgresql-support) in the `Reference Documentation` section of `HELP.md`.

**Describe the solution you'd like**

Create a PR to https://github.com/spring-io/start.spring.io to enable this feature.

**Describe alternatives you've considered**

N/A

**Additional context**
N/A

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Description Added
- [x] Expected solution specified

**Comment by saragluna** [comment]
We could refer to https://github.com/spring-io/start.spring.io/pull/946

**Comment by hui1110** [comment]
- When users choose MySQL + Azure, we should add MySQL reference documentation to HELP.md
- When users choose PostgreSQL + Azure, we should add PostgreSQL reference documentation to HELP.md

**Comment by hui1110** [comment]
@stliu I updated the description of this issue, please take a look.

**Comment by stliu** [comment]
> When the user selects PostgreSQL Driver and Azure Support in the Spring Initializer:
> 
> 2.1 The generated pom.xml will contain the following:
> 
>  <dependency>
>      <groupId>com.azure.spring</groupId>
>     <artifactId>spring-cloud-azure-starter-jdbc-postgresql</artifactId>
>  </dependency>
> spring-cloud-azure-starter and postgresql will be removed from pom.xml, they are already contained in spring-cloud-azure-starter-jdbc-postgresql.
> 

NO, this is not correct, we should just add the dependency `spring-cloud-azure-starter-jdbc-postgresql` when that two dependencies are chosen, DO NOT remove anything from the pom

**Comment by hui1110** [comment]
Closed this as completed in [9d2e80c ](https://github.com/spring-io/start.spring.io/commit/9d2e80c35a46dd92d093133aa3ecba90bd2a3828).
